### PR TITLE
Service file updates

### PR DESCRIPTION
**Pull request by eworm-de**
With latest systemd struggle we had to update systemd service ordering for our Arch Linux package. In the end we would like to use an upstream service file, but keep our security features.

Further fragmenting the service files is a bad idea I think... So updated the Fedora one. @jirka-h, please test carefully this works for Fedora...

### CODE REVIEW

File: init.d/service.fedora [hunk 1]
```diff
@@ -6,9 +6,12 @@ ConditionVirtualization=!container
 Before=sysinit.target shutdown.target systemd-journald.service
 
 [Service]
-ExecStart=/usr/sbin/haveged -w 1024 -v 1 --Foreground
+ExecStartPre=+/usr/bin/sysctl -w kernel.random.write_wakeup_threshold=1024
```
Comment (heftig) [inline init.d/service.fedora]: Shouldn't this be in `sysctl.d`?

Comment (eworm-de) [reply init.d/service.fedora]: Why `sysctl.d`? The executable is `/usr/bin/sysctl`.

Or do you want to drop a file to `/usr/lib/sysctl.d/`? That would alter the value if the package is installed but the service is not started.

**Comment by eworm-de** [comment]
Had some dependency issues after with my initial commits...

Pushed an update that should work.

Dropped two security features (`PrivateTmp=on` & `ProtectHome=on`) and reverted to use root, instead limited capabilities.

**Comment by jirka-h** [comment]
Thanks a lot for PR! I have tested it on Fedora 31 and it works reliably. Merging.

**Comment by jirka-h** [comment]
NoNewPrivileges=on 
breaks the service. 

haveged: Couldn't open random device: Permission denied

**Comment by eworm-de** [comment]
Are you sure your error is caused by `NoNewPrivileges=on`? The message is just about opening a file (`/dev/random`), nothing about elevating privileges. From `systemd.exec(5)`:

> Takes a boolean argument. If true, ensures that the service process and all its children can never gain new privileges through execve() (e.g. via setuid or setgid bits, or filesystem capabilities). [...]

**Pull request merged**